src/opm/io/eclipse/ERst.cpp

```diff
@@ -140,6 +139,15 @@ int ERst::getArrayIndex(const std::string& name, int number) const
     return std::distance(array_name.begin(), it);
 }
 
+int ERst::restartStepArrayIndex(const int seqnumValue) const
+{
+    auto pos = this->arrIndexRange.lower_bound(seqnumValue);
+    if (pos == this->arrIndexRange.end()) {
+        return -1;
+    }
+
+    return pos->second.first;
+}
```

Comment:
This should rather return the file offset directly - no reason to go via the position in a linear array.

Reply:
Good idea.  I'll update accordingly.